test/suite/Annotations.test.php

```diff
@@ -585,12 +590,149 @@ protected function testGetAnnotationsFromTrait()
+    {
+        if (version_compare(PHP_VERSION, '5.4.0', '<')) {
+            $this->pass();
+            return;
+        }
+
+        $annotations = Annotations::ofMethod('InsteadofTraitTester', 'baseTrait');
+        $this->check(count($annotations) === 2, 'baseTrait inherits annotation');
+        $this->check($annotations[0]->note === 'insteadof-base-trait-tester', 'inherited annotation goes first');
+        $this->check($annotations[1]->note === 'insteadof-base-trait-b', 'non-inherited annotation goes second');
+
+        $annotations = Annotations::ofMethod('InsteadofTraitTester', 'trate');
+        $this->check(count($annotations) === 2, 'trate inherits annotation');
+        $this->check($annotations[0]->note === 'insteadof-base-trait-tester', 'inherited annotation goes first');
+        $this->check($annotations[1]->note === 'insteadof-trait-a', 'non-inherited annotation goes second');
+    }
+
+    protected function testCanGetPropertyAnnotationsIncludedFromTrait()
+    {
+        if (version_compare(PHP_VERSION, '5.4.0', '<')) {
+            $this->pass();
+            return;
+        }
+
+        $annotations = Annotations::ofProperty(new \ReflectionClass('SimpleTraitTester'), 'sampleFromTrait');
+        $this->check(count($annotations) > 0, 'from class reflection and property name');
+
+        $annotations = Annotations::ofProperty(new \ReflectionProperty('SimpleTraitTester', 'sampleFromTrait'));
+        $this->check(count($annotations) > 0, 'from property reflection');
+
+        $annotations = Annotations::ofProperty(new SimpleTraitTester(), 'sampleFromTrait');
+        $this->check(count($annotations) > 0, 'from class object and property name');
+
+        $annotations = Annotations::ofProperty('SimpleTraitTester', 'sampleFromTrait');
+        $this->check(count($annotations) > 0, 'from class name and property name');
+
+        $annotations = Annotations::ofProperty('SimpleTraitTester', 'sampleFromAnotherTrait');
+        $this->check(count($annotations) > 0, 'from class name and property name from namespaced trait');
+    }
+
+    protected function testHandlesPropertyConflictWithTraits()
+    {
+        if (version_compare(PHP_VERSION, '5.4.0', '<')) {
+            $this->pass();
+            return;
+        }
+
+        set_error_handler(function ($errno, $errstring) { });
+        require_once __DIR__ . '/traits/property_conflict.php';
+        restore_error_handler();
+
+        $annotations = Annotations::ofProperty('PropertyConflictTraitTester', 'traitAndChild');
+        $this->check(count($annotations) === 1, 'traitAndChild does not inherit trait');
+        $this->check($annotations[0]->note === 'property-conflict-trait-tester', 'child annotation first');
+
+        $annotations = Annotations::ofProperty('PropertyConflictTraitTester', 'traitAndTraitAndParent');
+        $this->check(count($annotations) === 2, 'traitAndTraitAndParent inherits parent annotations');
```

Comment:
This is not right, because annotations from trait property & class property are never merged. Here trait property wins over parent class property with same name.